applications/asset_tracker/src/ui/led_pwm.c

```diff
@@ -136,13 +137,13 @@ static void work_handler(struct k_work *work)
 		int32_t next_delay =
 			leds.effect->steps[leds.effect_step].substep_time;
 
-		k_delayed_work_submit(&leds.work, K_MSEC(next_delay));
+		k_work_schedule(&leds.work, K_MSEC(next_delay));
 	}
 }
 
 static void led_update(struct led *led)
 {
-	k_delayed_work_cancel(&led->work);
+	k_work_cancel_delayable_sync(&led->work, &led->work_sync);
```

Comment:
This isn't new, so shouldn't block the change, but this doesn't look right.  The check at 151 suggests that disabling led updates involves setting `led->effect` to NULL.  However, the work handler unconditionally dereferences `led->effect`.  If the cancel fails (or if whatever sets `led->effect` to NULL has another race with the work handler) the handler could cause a fault.

Reply:
Thanks. I have added  a new task to investigated this: CIA-317